\documentclass{article}
\usepackage[dvipsnames]{xcolor}
\usepackage{caption}
\usepackage{graphicx}
\usepackage[hidelinks]{hyperref}
\usepackage{ragged2e}
\usepackage[margin=0.8in]{geometry}
\usepackage{lmodern}
\renewcommand*\familydefault{\sfdefault}
\usepackage[T1]{fontenc}
\usepackage{fancyhdr}
\setlength\footskip{40pt} 
\pagestyle{fancy}



\renewcommand{\headrulewidth}{2pt}
\renewcommand{\headrule}{\hbox to \headwidth{\color{BurntOrange}\leaders\hrule height \headrulewidth\hfill}}
\renewcommand{\footrulewidth}{0.5pt}


\fancyfoot[R]{ % right
   \includegraphics[scale=0.2]{CGS_logo_final.jpg}
}
\fancyfoot[L]{ % left
   \includegraphics[scale=0.2]{UNI_logo.jpg}
}

\title{\color{BurntOrange}{\textbf{Microarray Gene Expression Report}}}
\author{\includegraphics[scale=0.4]{CGS_logo_final.jpg}}
\color{black}

\begin{document}
<< LoadLibraries,echo=FALSE, warning=FALSE, message=FALSE>>=
library(limma)
library(lumi)
@
\maketitle
\tableofcontents
\color{BurntOrange}
\section{\textbf{Material and Methods}}

\subsection{Array processing}
\color{black}

RNA was assessed for concentration and quality using a SpectroStar (BMG Labtech, Aylesbury, UK) and a Bioanalyser (Agilent Technologies, Cheadle, UK). Microarray experiments were performed at Cambridge Genomic Services, University of Cambridge, using a species specific Gene 2.1 ST Array Plate (Affymetrix, Wooburn Green, UK) in combination with WT PLUS amplification kit (Affymetrix) according to the manufacturer's instructions. Briefly, 100ng Total RNA was amplified along with inline PolyA spike in control RNA, using the WT PLUS amplification kit (Affymetrix). Successfully amplified samples were labelled using the GeneChip WT terminal labelling kit (Affymetrix) using the in line hybridization controls. Plate arrays were processed on the GeneTitan instrument (Affymetrix) using the GeneTitan Hybridization, Wash and Stain kit (Affymetrix). Samples were hybridized to the array, washed, stained and scanned using the array specific parameters provided by Affymetrix. Finally basic visual quality control was performed using Command Console Viewer (Affrymetrix) prior to bioinformatic quality control.

\color{BurntOrange}
\subsection{Bioinformatics}
\color{black}
After scanning the files generated by the scanner (CEL files) are loaded in R using the oligo package from bioconductor \cite{oligo}. No background correction or normalisation is applied at this stage. In order to assess the quality of the data plots of the control probes are generated along with boxplot, MAplot and intensity distribution plot. The raw data are then processed using the RMA method \cite{rma}. The data are background corrected, normalised using quantile and summarized. Once the data are processed, the comparisons are performed using the limma package with results corrected for multiple testing using False Discovery Rate (FDR) testing. 
In the interest of reproducibility the packages version and R session information \cite{R} are provided in the reference.
\pagebreak
\color{BurntOrange}
\section{Technical QC report}
\color{black}
This section focuses on the technical aspect of the array, using the Affymetrix controls integrated to the array to check if the data generated is of good quality

\color{BurntOrange}
\subsection{BAC spyke control}
\color{black}
This control uses BAC spykes, they hybridize to the pre-labeled bacterial spike controls (BioB, BioC, BioD, and Cre). This control is useful to identify problems with the hybridization and/or chip. Metrics in this control category will have more variability than other categories (i.e., positive controls) due to the limited number of spikes and probesets used. If the bacterial spikes display the expected rank order (BioB<BioC<BioD<Cre), then the problem is likely to be prior to the hybridization phase.

\begin{center}
\begin{minipage}{\linewidth}
\makebox[\linewidth][c]{\href{run:controls/control_spykes.pdf}{\includegraphics[scale=0.5]{controls/control_spykes.pdf}}}
\captionof{figure}{Bacterial spykes controls}
\end{minipage}
\end{center}
\vspace{10mm}



\pagebreak
\color{BurntOrange}
\subsection{PolyA spykes controls}
\color{black}
This control consists of the set of polyadenylated RNA spikes (Lys, Phe, Thr, and Dap). This control category is useful to identify problems with the target preparation. Metrics in this category will have more variability than other categories (i.e., positive controls, all probesets) due to the limited number of spikes and probesets for this category.
If the polyA spikes display the expected rank order (lys<phe<thr<dap) then the problem is likely to be with the RNA sample or the target preparation.

\begin{center}
\begin{minipage}{\linewidth}
\makebox[\linewidth]{\href{run:controls/control_polyA.pdf}{\includegraphics[scale=0.5]{controls/control_polyA.pdf}}}
\captionof{figure}{PolyA spykes controls}
\end{minipage}
\end{center}
\vspace{10mm}


\pagebreak
\color{BurntOrange}
\subsection{Positive and negative controls}
\color{black}
Negative controls: is the set of putative intron based probe sets from putative housekeeping genes. Specifically, a number of species specific probesets on 3' IVT arrays were shown to have constitutive expression over a large number of samples. The genes for these probesets were identified and multiple four probe probesets were selected against the putative intronic regions.Thus in any given sample some (or many) of these putative intronic regions may be transcribed and retained. Furthermore, some (or many) of the genes may not be constitutive within certain data sets. These caveats aside, this collection makes for a moderately large collection of probesets which in general have very low signal values.
Positive controls: is the set of putative exon based probe sets from putative housekeeping genes. Specifically, a number of species specific probesets on 3'IVT arrays were shown to have constitutive expression over a large number of samples. The genes for these probesets were identified and multiples of four probe probesets were selected against the putative exonic regions.Thus in any given sample some (or many) of these putative exonic regions may not be transcribed or may be spliced out. Furthermore, some (or many) of the genes may not be constitutive within certain data sets. These caveats aside, this collection makes for a moderately large collection of probesets with target present which in general have moderate to high signal values. 

These probesets are used to estimate the true positive rate for the pos\_vs\_neg\_auc metric. The pos\_control and all\_probeset categories are useful in getting a handle on the overall quality of the data from each chip. Metrics based on these categories will reflect the quality of the whole experiment (RNA, target prep, chip, hybridization, scanning, and griding) and the nature of the data being used in downstream statistical analysis.

\begin{center}
\begin{minipage}{\linewidth}
\makebox[\linewidth]{\href{run:controls/control_positive_negative.pdf}{\includegraphics[scale=0.5]{controls/control_positive_negative.pdf}}}
\captionof{figure}{Positive/negative controls}
\end{minipage}
\end{center}
\vspace{10mm}



\pagebreak
\color{BurntOrange}
\subsection{NUSE boxplot}
\color{black}
NUSE stands for Normalised Unscalled Standard Error, it is the fitting of individual probe error in the Probe Level Model (PLM). The PLM models the expression measures using a M-estimator robust regression, in other word it fits a model to the probe-intensity data. Here we use the standard error of this model to check the sample quality. The NUSE values are standardized at the probe level accros the arrays, median values for each probe-set are set to 1. The boxplot is usefull to check that all the distributions are centered near 1. An array with a boxplot centered around 1.1 is flagged as bad quality. If one array show a higher spread of NUSE distribution, it's standard  error is higher than the rest and it can be a sign of bad quality.
\begin{center}
\begin{minipage}{\linewidth}
  \makebox[\linewidth]{\href{run:controls/control_NUSE.pdf}{\includegraphics[scale=0.6]{controls/control_NUSE.pdf}}}
	\captionof{figure}{NUSE boxplot}
\end{minipage}
\end{center}
\vspace{10mm}


\pagebreak
\color{BurntOrange}
\subsection{RLE boxplot}
\color{black}
The Relative Log Expression values are calculated for each probe-set by taking the ratio between the expression of a probe-set and the median expression of this probe-set accross all the arrays of the experiment. It is assumed that the majority of a probe-set will have similar levels accross the arrays, so the ratio should be centered on 0 on the log scale. If the boxplot are not centered on 0 or have a different spread compare to the other samples, it is a sign of low quality.

\begin{center}
\begin{minipage}{\linewidth}
  \makebox[\linewidth]{\href{run:controls/control_RLE.pdf}{\includegraphics[scale=0.6]{controls/control_RLE.pdf}}}
  \captionof{figure}{RLE boxplot}
\end{minipage}
\end{center}
\vspace{10mm}



\pagebreak
\color{BurntOrange}
\subsection{Intensity Boxplots}
\color{black}
Boxplots show the distribution of the intensity as a box with whiskers, the bottom of the box delimits the 25th percentile, the middle of the box the median, the top of the box the 75th percentile. Samples should have similar distribution of intensity. The intensity scale is log base 2.

\begin{center}
\begin{minipage}{\linewidth}
	\makebox[\linewidth]{\href{run:raw_data_plots/boxplot_raw_groups.pdf}{\includegraphics[scale=0.6]{raw_data_plots/boxplot_raw_groups.pdf}}}
	\captionof{figure}{Boxplot}
\end{minipage}
\end{center}
\vspace{10mm}


\pagebreak
\color{BurntOrange}
\subsection{Density}
\color{black}
The density plot represents the distribution of the intensity by plotting  the intensities on the x axis against the frequency on the y axis, it is a less schematic representation of the distribution of the intensities. The distribution should be fairly similar for all the arrays.
\begin{center}
\begin{minipage}{\linewidth}
	\makebox[\linewidth]{\href{run:raw_data_plots/density_plot_raw_data.pdf}{\includegraphics[scale=0.6]{raw_data_plots/density_plot_raw_data.pdf}}}
	\captionof{figure}{Density plot}
\end{minipage}
\end{center}
\vspace{10mm}

\pagebreak
\color{BurntOrange}
\subsection{GC content and intensity}
\color{black}
Array intensity has been shown to be dependent to GC content of the probes. To estimate the strength of this effect, the intensity for different GC content of the probe are ploted as boxplos. If no strong increase in intensisty occurs with increasing GC content it means that the impact of GC content on the overall intensity is not significant. 


<<GetGCContent, echo=FALSE,results="asis">>=
current=getwd()
reportPath="raw_data_plots/"
gcplots=list.files(path=reportPath,pattern="GCContent")
cat("\\begin{center}\n")
for(i in 1:length(gcplots)){
 
  cat("\\begin{minipage}{\\linewidth}\n")
  cat(paste("\\makebox[\\linewidth]{\\href{run:raw_data_plots/",gcplots[i],"}{\\includegraphics[scale=0.6]{raw_data_plots/",gcplots[i],"}}}\n",sep=""))
  cat("\\end{minipage}\n")
  
}
cat("\\end{center}\n")
@
\vspace{10mm}

\pagebreak
\color{BurntOrange}
\subsection{MA plot}
\color{black}
MA plot is used to show variations in intensity for a sample against a pseudo sample constructed using the intensity of all the samples. M (y axis) is the log ratio of intensity between the sample and the pseudo sample and A (x axis) is the mean average of intensity.  Ideally the data will be close to the y=0 line. It is expected that the data will display more noise (deviation from the y=0 line) at low intensity than a high intensity. Outlier samples for technical reasons can be spotted as they might show a large deviation from the central line, a large cloud of data spreading far from the  y=0 for the whole intensity range. Trend in the data can also be seen using this plot ("banana" shaped plot), usually indicative of technical variation between sample. This type of trend should be corrected by the normalisation.
<<GetMAPlot, echo=FALSE,results="asis">>=
current=getwd()
reportPath="raw_data_plots/"
maplots=list.files(path=reportPath,pattern="MAplot")
for(i in 1:length(maplots)){
  cat("\\begin{center}\n")
  cat("\\begin{minipage}{\\linewidth}\n")
  cat(paste("\\makebox[\\linewidth]{\\href{run:raw_data_plots/",maplots[i],"}{\\includegraphics[scale=0.6]{raw_data_plots/",maplots[i],"}}}\n",sep=""))
  cat("\\end{minipage}\n")
  cat("\\end{center}\n")
}
@

\pagebreak
\color{BurntOrange}
\section{Normalisation}
\subsection{Normalisation Method}
\color{black}
The processing of the data to generate normalised intensity is done using Robust Multichip Analysis \cite{rma}. This method sumarize the intensities of all probes covering a gene, constructing a model containing a "probe" and a "chip" effect. The advantage of this model is in the use of all the array available to give a more robust model and better normalisation across the whole experiment. All the arrays will be normalised together as the robustness of the normalisation increases when using more chips.
The probes are also filtered according to the intensity level, by default at least 2 samples must have probes with signal above the median for whole set of probe set, e.g. all the probes on the array.Given the number of probes, it is likely that a good number of them will not be above background. Therefore using the median intensity level should give us a good estimator to identify medium to highly expresed probe. Keeping to many probes when doing the comparison will have detrimental effect on power as it increase the stringency of the False Discovery Rate multiple testing correction. 


\color{BurntOrange}
\subsection{Normalisation  QC}
\color{black}
All the QC graphs that were generated for the raw data are also regenerated for the normalisation data in order to check if there are any issues caused by normalisation. 

\color{BurntOrange}
\subsection{Intensity Boxplots}
\color{black}
Boxplots show the distribution of the intensity as a box with whiskers, the bottom of the box delimits the 25th percentile, the middle of the box the median, the top of the box the 75th percentile. Samples should have similar distribution of intensity. The intensity scale is log base 2.

\begin{center}
\begin{minipage}{\linewidth}
  \makebox[\linewidth]{\href{run:rma_plots/boxplot_rma_core.pdf}{\includegraphics[scale=0.6]{rma_plots/boxplot_rma_core.pdf}}}
	\captionof{figure}{Boxplot}
\end{minipage}
\end{center}
\vspace{10mm}

\pagebreak
\color{BurntOrange}
\subsection{Density}
\color{black}
The density plot represents the distribution of the intensity by plotting  the intensities on the x axis against the frequency on the y axis, it is a less schematic representation of the distribution of the intensities.

\begin{center}
\begin{minipage}{\linewidth}
  
	\makebox[\linewidth]{\href{run:rma_plots/density_plot_rma_data.pdf}{\includegraphics[scale=0.4]{rma_plots/density_plot_rma_data.pdf}}}
	\captionof{figure}{Density plot}
\end{minipage}
\end{center}

\vspace{10mm}

\pagebreak
\color{BurntOrange}
\subsection{MA plot}
\color{black}
MA plot is used to show variations in intensity for a sample against a pseudo sample constructed using the intensity of all the samples. M (y axis) is the log ratio of intensity between the sample and the pseudo sample and A (x axis) is the mean average of intensity.  Ideally the data will be close to the y=0 line. It is expected that the data will display more noise (deviation from the y=0 line) at low intensity than a high intensity. Outlier samples for technical reasons can be spotted as they might show a large deviation from the central line, a large cloud of data spreading far from the  y=0 for the whole intensity range. Trend in the data can also be seen using this plot ("banana" shaped plot), usually indicative of technical variation between sample. This type of trend should be corrected by the normalisation.
<<GetMAPlotRMA, echo=FALSE,results="asis">>=
current=getwd()
reportPath="rma_plots/"
maplots=list.files(path=reportPath,pattern="MAplot")
for(i in 1:length(maplots)){
  cat("\\begin{center}\n")
  cat("\\begin{minipage}{\\linewidth}\n")
  cat(paste("\\makebox[\\linewidth]{\\href{run:rma_plots/",maplots[i],"}{\\includegraphics[scale=0.6]{rma_plots/",maplots[i],"}}}\n",sep=""))
  cat("\\end{minipage}\n")
  cat("\\end{center}\n")
}
@

\pagebreak
\color{BurntOrange}
\section{Sample Groups QC}
\color{black}

In this section the relationship between the samples is assessed. We generate Multi dimensional scalling and hierarchical clustering to look at how to sample relate to their group and how the groups relate to each others. This relationship will be assessed before and after rma processing.

\color{BurntOrange}
\subsection{MultiDimensional Scaling (MDS) clustering}
\color{black}
MDS clustering clusters the samples in two dimensions (or more) instead of one, decomposing the variance into several components. Two graphs are generated, on with 2 dimensions and one with 3 dimensions. MDS can identify sources of variation potentially missed by relying on hierachical clustering alone. Note that individual MDS clustering is available for each group in \href{run:raw_data_plots/}{raw\_data\_plots\_folder} for the raw data and after normalisation in \href{run:rma_plots/}{rma\_plots folder}.
\begin{center}
\begin{minipage}{\linewidth}
  \makebox[\linewidth]{\href{run:raw_data_plots/MDS_clustering_raw_data.pdf}{\includegraphics[scale=0.6]{raw_data_plots/MDS_clustering_raw_data.pdf}}}
	\captionof{figure}{MDS clustering before normalisation}
\end{minipage}
\end{center}
\begin{center}
\begin{minipage}{\linewidth}
  \makebox[\linewidth]{\href{run:raw_data_plots/MDS_clustering_raw_data_3D.pdf}{\includegraphics[scale=0.6]{raw_data_plots/MDS_clustering_raw_data_3D.pdf}}}
  \captionof{figure}{3D MDS clustering before normalisation}
\end{minipage}
\end{center}
\begin{center}
\begin{minipage}{\linewidth}
  \makebox[\linewidth]{\href{run:rma_plots/MDS_clustering_rma_data.pdf}{\includegraphics[scale=0.6]{rma_plots/MDS_clustering_rma_data.pdf}}}
  \captionof{figure}{MDS clustering after rma processing}
\end{minipage}
\end{center}
\begin{center}
\begin{minipage}{\linewidth}
  \makebox[\linewidth]{\href{run:rma_plots/MDS_clustering_norm_data_3D.pdf}{\includegraphics[scale=0.6]{rma_plots/MDS_clustering_norm_data_3D.pdf}}}
  \captionof{figure}{3D MDS clustering after rma processing}
\end{minipage}
\end{center}
\vspace{10mm}


\pagebreak
\color{BurntOrange}
\subsection{Hierarchical clustering}
\color{black}
 Hierarchical clustering should show that sample cluster with their given group. This can be used to spot potential sample swaps. If all the groups are mixed together, it is an indication that they might be little changes between the  groups. Two graphs are displayed below, one simple tree the other is a coloured tree. The colour of the sample matches the group they belong to, while the colour on the tree show the different cluster. Ideally the colours between the two should match, showing that the sample of a given group cluster together.
Hierachical clustering graphs for individual groups are available in the \href{run:raw_data_plots/}{raw\_data\_plots folder} for the raw data and in the \href{run:rma_data_plots/}{rma\_data\_plots folder} for the normalised data.
\begin{center}
\begin{minipage}{\linewidth}
  \makebox[\linewidth]{\href{run:raw_data_plots/Hierarchical_clustering_raw_data.pdf}{\includegraphics[scale=0.6]{raw_data_plots/Hierarchical_clustering_raw_data.pdf}}}
  \captionof{figure}{Hierarchical clustering before normalisation}
\end{minipage}
\end{center}
\begin{center}
\begin{minipage}{\linewidth}
  \makebox[\linewidth]{\href{run:raw_data_plots/Hierarchical_clustering_raw_data_coloured.pdf}{\includegraphics[scale=0.6]{raw_data_plots/Hierarchical_clustering_raw_data_coloured.pdf}}}
  \captionof{figure}{Hierarchical clustering before normalisation}
\end{minipage}
\end{center}
\begin{center}
\begin{minipage}{\linewidth}
  \makebox[\linewidth]{\href{run:rma_plots/Hierarchical_clustering_norm_data.pdf}{\includegraphics[scale=0.6]{rma_plots/Hierarchical_clustering_norm_data.pdf}}}
  \captionof{figure}{Hierachical clustering after rma processing}
\end{minipage}
\end{center}
\begin{center}
\begin{minipage}{\linewidth}
  \makebox[\linewidth]{\href{run:rma_plots/Hierarchical_clustering_norm_data.pdf}{\includegraphics[scale=0.6]{rma_plots/Hierarchical_clustering_norm_data_coloured.pdf}}}
  \captionof{figure}{Hierachical clustering after rma processing}
\end{minipage}
\end{center}
\vspace{10mm}



\pagebreak
\color{BurntOrange}
\section{\textbf{Comparisons}}
\subsection{How to interpret your results}
\color{black}
This section contains information about the comparisons performed.

We regenerate the hierachical clustering and MDS clustering for the groups of being compared. A 3 dimensions MDS plot is also available in the \href{run:results/graph/}{results/graph folder}  if the number of samples allowed for its generation.
A summary table is generated (see below) to give an overview of the results. For each comparison a table with the top 10 genes is displayed along with a volcano plot. The volcano plot is useful to get an overview of the level of change and the level of significance, the x axis shows the level of change, the fold change (log2 scale), while the y axis represent the significance, -log10(pvalue). 
A direct link to the table is provided for each comparison.

To work on the results there are 2 important columns to use, logFC and adj.pvalue. The adj.pvalue contains the p-value corrected for multiple testing using False Dicovery Rate. The p.value column should \textbf{not} be used as it is not corrected for multiple testing, resulting in a large number of false positives.
The adjusted p-value should be used as a primary filtering column. If there are still too many results after filtering using "classic" levels (0.05 or 0.01) consider lowering it further or using the logFC column as a secondary fiter to look at the higher rates of change in your data. The logFC represents the log in base 2 of the fold change. A value of 1 is equivalent to a change of 2 folds, 2 is 4 folds, 3 is 8, etc... Using the value as a power of 2 gives you the real fold change.

If there are still a large number of probes to study, it is recommended to take a functional approach to analyse the results. The aim of a functional analysis is to group genes according to function/pathways in order to reduce the complexity of the set. For this Gene Ontology tools or pathway finder can be used.

\textbf{A note on the p-value:} in some case the adjusted p-value will have the same value for several probes, which might look like an error has occured during the correction. This can happen as a results of how the correction is performed. The steps are as follow:
\begin{itemize}
\item Sort the p-value from low (rank 1) to high (rank N, with N the number of comparisons performed)
\item Multiply the p-value by N
\item Divide the p-value by its rank
\item Check that the rank is kept
\end{itemize}
The last point is the reason why the adjusted p-value can share the same value, if the ranks are not kept and a higher ranked probe ends up with a lower adjusted p-value than a lower rank probe, the lower value of the higher ranked probe will be used for the lower ranked p-value.
\color{BurntOrange}
\subsection{Comparisons summary}
\color{black}

<<Summary table, echo=FALSE, results='asis', warning=FALSE>>=
library(xtable)
summary=read.table("results/summary_comparisons.txt", sep="\t", header=T)
print(xtable((summary[,c(1:3)])), include.rownames=FALSE,
caption.placement = getOption("xtable.caption.placement", "top"))
@

\pagebreak
\color{BurntOrange}
\subsection{Individual Comparisons}
\color{black}

<<Get comparisons,echo=FALSE, results='asis'>>=
 path="results"
 setwd(path)
 listResults=list.files(recursive = T)
 listComp=listResults[grep(listResults,pattern="tables")]
 listGraphs=listResults[grep(listResults, pattern="graphs")]

 for( i in 1:length(listComp)){
   comp=gsub("_"," ",listComp[i])   
   comp=gsub("tables/","",comp)
   comp=gsub(".csv","",comp)
   cat("\\color{BurntOrange}")
   cat(paste("\\subsubsection{",comp,"}\n",sep=""))
   cat("\\color{black}")
   comparisonsTable=paste("/results/",listComp[i],sep="")
  
   sampleTable=listComp[i]
   summary=read.table(sampleTable, sep=",", header=T,nrows = 11)
   gene_info=summary[c(1:10),8]
   for(l in 1:length(gene_info)){
     gene<-strsplit(as.character(gene_info[l]),split="//")[[1]][2]
     summary[l,1]<-gene
   }
   print(xtable((summary[c(1:10),c(1,3,5,6,19,23)])), include.rownames=FALSE,
        
   caption.placement = getOption("xtable.caption.placement", "top"))
   cat("\\centering")
   cat(paste("\\href{",comparisonsTable,"}{Click here to open the comparison table}\n\n", sep=""))
   cat("\\justify")
   compNames=gsub(pattern="tables/",listComp[i],replacement="")
   compNames=gsub(pattern=".csv",compNames, replacement="")
   compNames=gsub(pattern="results_",compNames, replacement = "")
   volcano=listGraphs[grep(pattern=paste0("Volcano_",compNames), listGraphs)]
   compNames=gsub(pattern="_vs_", compNames, replacement="_v_")
   hc=listGraphs[grep(pattern=paste0("Hierarchical_clustering_rma_data_",compNames),listGraphs)]
   mds=listGraphs[grep(pattern=paste0("MDS_clustering_rma_data_",compNames), listGraphs)]
  
   
   cat("\\begin{minipage}{\\linewidth}")
   cat(paste("\\makebox[\\linewidth]{\\href{run:results/",volcano,"}{\\includegraphics[height=8cm, width=10cm]{results/",volcano,"}}}\n",sep=""))
   cat("\\captionof{figure}{Volcano Plot}\n") 
   cat("\\end{minipage}\n")
  

    cat("\\begin{minipage}{\\linewidth}")
   cat(paste("\\makebox[\\linewidth]{\\href{run:results/",hc,"}{\\includegraphics[height=8cm, width=10cm]{results/",hc,"}}}\n",sep=""))
   cat("\\captionof{figure}{Hierarchical clustering of the two groups being compared}\n") 
   cat("\\end{minipage}\n")
  
   
cat("\\begin{minipage}{\\linewidth}")
   cat(paste("\\makebox[\\linewidth]{\\href{run:results/",mds,"}{\\includegraphics[height=8cm, width=10cm]{results/",mds,"}}}\n",sep=""))
   cat("\\captionof{figure}{Hierarchical clustering of the two groups being compared}\n") 
   cat("\\end{minipage}")
   cat("\\pagebreak\n")
}

@


\bibliographystyle{plain}
%\bibliography{biblio}
\begin{thebibliography}{9}
\bibitem{oligo}
%\Sexpr{as.character(lumiCit)}
<<biblio lumi,echo=FALSE,results='asis'>>=
citation("oligo")
cat(paste("Package version: ",packageVersion("oligo"),sep=""))
@

\bibitem{rma}
Rafael. A. Irizarry, Benjamin M. Bolstad, Francois Collin, Leslie M. Cope, Bridget Hobbs and Terence P. Speed (2003), Summaries of Affymetrix GeneChip probe level data Nucleic Acids Research 31(4):e15

\bibitem{limma}
<<biblio limma,echo=FALSE,results='asis'>>=
citation("limma")[1]
cat(paste("Package version: ",packageVersion("limma"),sep=""))
@
\bibitem{R}
\begin{verbatim}
<<R info,echo=FALSE,results='asis'>>=
sessionInfo()
@
\end{verbatim}

\end{thebibliography}

\end{document}